src/08_ecosystem/00_chapter.md

```diff
@@ -0,0 +1,83 @@
+A common choice is to use utilities from `futures` with an executor from another crate.
+
+### Popular Async Runtimes
+There is no asynchronous runtime in the standard library, and none are officially recommended.
+The following crates provide popular runtimes.
+- [Tokio](https://docs.rs/tokio/): A popular async ecosystem with HTTP, gRPC, and tracing frameworks.
+- [async-std](https://docs.rs/async-std/): A crate that provides asynchronous counterparts to standard library components.
+- [smol](https://docs.rs/smol/): A small, simplified async runtime.
+Provides the `Async` trait that can be used to wrap structs like `UnixStream` or `TcpListener`.
+- [fuchsia-async](https://fuchsia.googlesource.com/fuchsia/+/master/src/lib/fuchsia-async/):
+An executor for use in the Fuchsia OS.
+
+## Determining Ecosystem Compatibility
+Not all async applications, frameworks, and libraries are compatible with each other, or with every OS or platform.
+Most async code can be used with any ecosystem, but some frameworks and libraries require the use of a specific ecosystem.
+Ecosystem constraints are not always documented, but there are several rules of thumb to determine
+whether a library, trait, or function depends on a specific ecosystem.
+
+Any async code that interacts with async I/O, timers, interprocess communication, or tasks
+generally depends on a specific async executor or reactor.
+All other async code, such as async expressions, combinators, synchronization types, and streams
+are usually ecosystem independent, provided that any nested futures are also ecosystem independent.
+Before beginning a project, it's recommended to research relevant async frameworks and libraries to ensure
+compatibility with your chosen runtime and with each other.
+
+Notably, `Tokio` uses the `mio` reactor and defines its own versions of async I/O traits,
+including `AsyncRead` and `AsyncWrite`.
+On its own, it's not compatible with `async-std` and `smol`,
+which rely on the [`async-executor` crate](https://docs.rs/async-executor), and the `AsyncRead` and `AsyncWrite`
+traits defined in `futures`.
+
+Conflicting runtime requirements can sometimes be resolved by compatibility layers
+that allow you to call code written for one runtime within another.
+For example, the [`async_compat` crate](https://docs.rs/async_compat) provides a compatibility layer between
+`Tokio` and other runtimes.
+
+Libraries exposing async APIs should not depend on a specific executor or reactor,
+unless they need to spawn tasks or define their own async I/O or timer futures.
+Ideally, only binaries should be responsible for scheduling and running tasks.
+
+## Single Threaded vs Multi-Threaded Executors
+Async executors can be single-threaded or multi-threaded.
+For example, the `async-executor` crate has both a single-threaded `LocalExecutor` and a multi-threaded `Executor`.
+
+A multi-threaded executor makes progress on several tasks simultaneously.
```

Comment:
nit:

```suggestion
A multi-threaded executor makes progress on several tasks in parallel.
```

since simultaneously could also be taken to mean "concurrently" which is also provided by single-threaded runtimes.